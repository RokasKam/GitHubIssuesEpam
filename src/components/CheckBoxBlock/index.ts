import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };