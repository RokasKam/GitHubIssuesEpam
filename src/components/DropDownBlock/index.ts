import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };