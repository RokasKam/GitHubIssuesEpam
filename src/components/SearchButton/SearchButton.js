import React from 'react'
import { Text, TouchableOpacity, } from 'react-native';
import { styles } from './SearchButton.styles';

export const SearchButton = ({ onPress, text }) => {

    return (
        <TouchableOpacity
            style={styles.loginButton}
            onPress={() => onPress()}>
            <Text style={styles.loginText}>{text}</Text>
        </TouchableOpacity>
    )
};

