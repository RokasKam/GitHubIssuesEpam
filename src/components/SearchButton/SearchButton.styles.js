import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    loginButton: {
        alignItems: "center",
        justifyContent: "center",
        height: 50,
        width: "50%",
        borderRadius: 30,
        marginLeft: 10,
        backgroundColor: "#323C4C",
    },
    loginText: {
        color: "white",
    },
});