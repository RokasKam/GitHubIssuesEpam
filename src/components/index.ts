export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

