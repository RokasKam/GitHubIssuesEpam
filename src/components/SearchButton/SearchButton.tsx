import React from 'react'
import { Text, TouchableOpacity, } from 'react-native';
import { styles } from './SearchButton.styles';
import { Props } from './SearchButton.type';

export const SearchButton: React.FC<Props> = ({ onPress, text }) => {

    return (
        <TouchableOpacity
            style={styles.loginButton}
            onPress={onPress}>
            <Text style={styles.loginText}>{text}</Text>
        </TouchableOpacity>
    )
};

