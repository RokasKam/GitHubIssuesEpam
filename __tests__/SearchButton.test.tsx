import React from 'react';
import renderer from 'react-test-renderer';
import { SearchButton } from '../src/components/SearchButton'

test('renders correctly', () => {
  const tree = renderer.create(<SearchButton onPress={function (): void {
    throw new Error('Function not implemented.');
  }} text={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});