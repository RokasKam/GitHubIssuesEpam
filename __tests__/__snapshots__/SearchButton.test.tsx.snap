// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#323C4C",
      "borderRadius": 30,
      "height": 50,
      "justifyContent": "center",
      "marginLeft": 10,
      "opacity": 1,
      "width": "50%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  />
</View>
`;
